docs/DeveloperGuide.md

```diff
@@ -1,34 +1,956 @@
+
+This section explains the implementation and design considerations for managing `Notebook`s.
+
+##### 4.3.1.1 Implementation
+There are two main functions in notebook management: add and remove.
+<br>
+
+Given below is an example usage scenario and how the add notebook function behaves.
+
+Step 1. The user launches the application for the first time. `CliUserInterface#executeCommand` is called when the user adds new `Notebook` into the `NotebookShelf`.
+
+Step 2. The user types `add /nCS2113T`. The `add` command is passed through `InputParser#getCommandFromInput`.
+
+Step 3. `InputParser#parseNotebookTitle` is then called, and it returns the `titleToAdd`, which is `CS2113T`.
+
+Step 4. A constructor for `AddCommandNotebookMode` is created.
+
+Step 5. `AddCommandNotebookMode#execute()` runs, which then calls `NotebookShelf#addNotebook`.
+
+Step 6. A new `Notebook`, entitled `CS2113T` is initialised.
+
+The UML sequence diagram below shows how the add task command works.
+![Sequence Diagram for Add Task Command](/diagrams/class/jpeg/add_notebook.jpg)
+
+<hr>
+
+The figure below shows how the "remove task" command works:
+<img src="https://user-images.githubusercontent.com/60319628/96821973-9176e600-145b-11eb-95b7-5bf885ea1867.png">
+
+After calling `InputParser#getCommandFromInput` from `CliUserInterface`:
+1. `InputParser` parses the input to return the `notebookTitleToRemove`, `sectionTitleToRemove` and `pageNumberToRemove`.
+Some of these members may be empty.
+2. `InputParser` constructs and returns the `RemoveCommandNotebookMode` class with constructor as shown below:
+```java
+public RemoveCommandNotebookMode(String notebookTitle, String sectionTitle,
+                                     int pageNumber, AppState appState) {
+    this.appState = appState;
+    notebookTitleToRemove = notebookTitle;
+    sectionTitleToRemove = sectionTitle;
+    pageNumberToRemove = pageNumber;
+
+    currentBookshelf = appState.getCurrentBookShelf();
+    currentNotebook = appState.getCurrentNotebook();
+    currentSection = appState.getCurrentSection();
+}
+```
+3. Method **execute()** is called by `CliUserInterface` to delete a notebook, section, or page, depending on the input.
+A switch-case block is used to determine the method to call based on the `appMode`.
+4. If the deletion is successful, `CliMessages` displays the message to the user.
+
+#### 4.3.2. Tag Feature
+
+The user can add a `tag` to a `Notebook`, `Section` or `Page`. This section describes the implementation and design
+considerations for this feature.
+
+##### 4.3.2.1. Implementation
+The `Notebook`, `Section` and `Page` classes each contain a member `tag` of type String.
+
+The figure below shows how the tag operation works:
+
+![Sequence Diagram for Tag Notebook command](/diagrams/class/jpeg/SequenceDiagram_TagNotebook.jpg)
+
+There are 3 crucial processes during the tag operation. When the user enters `tag /tCS2113T` into the command window while using the application:
```

Comment:
Would it be possible to separate the sequence diagrams into the 3 processes mentioned here? Right now it's quite hard to understand the diagram because the text in the diagram is quite small (opening it as a new image doesn't help). You could also consider breaking it into `ref` frames too.

Reply:
Agreed, it does look quite pixelated and hard to read. We'll try to break it up into smaller parts.